src/main/java/com/kantenkugel/discordbot/versioncheck/changelog/JenkinsChangelogProvider.java

```diff
@@ -0,0 +1,126 @@
+package com.kantenkugel.discordbot.versioncheck.changelog;
+
+import com.kantenkugel.discordbot.jenkinsutil.JenkinsApi;
+import com.kantenkugel.discordbot.jenkinsutil.JenkinsBuild;
+import com.kantenkugel.discordbot.jenkinsutil.JenkinsChange;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class JenkinsChangelogProvider implements ChangelogProvider
+{
+    private final JenkinsApi jenkins;
+    private final String githubUrl;
+
+    public JenkinsChangelogProvider(JenkinsApi jenkins, String githubUrl)
+    {
+        this.jenkins = jenkins;
+        this.githubUrl = githubUrl.endsWith("/") ? githubUrl : githubUrl + '/';
+    }
+
+    @Override
+    public String getChangelogUrl()
+    {
+        return jenkins.getChangesetUrl();
+    }
+
+    @Override
+    public boolean supportsIndividualLogs()
+    {
+        return true;
+    }
+
+    @Override
+    public Changelog getChangelog(String version)
+    {
+        List<Changelog> changelogs = getChangelogs(version, version);
+        return changelogs.isEmpty() ? null : changelogs.get(0);
+    }
+
+    @Override
+    public List<Changelog> getChangelogs(String startVersion, String endVersion)
+    {
+        int start = extractBuild(startVersion);
+        int end = extractBuild(endVersion);
+        if(start == 0 || end == 0)
+            return Collections.emptyList();
+        if(start > end)
+        {
+            int tmp = start;
+            start = end;
+            end = tmp;
+        }
+
+        List<Changelog> changelogs = new ArrayList<>();
+        for (int i = start; i <= end; i++)
+        {
+            JenkinsBuild build = jenkins.getBuild(i);
+            if(build == null)
+                continue;
+
+            String title = build.status == JenkinsBuild.Status.SUCCESS
+                    ? build.artifacts.values().iterator().next().fileNameParts.get(1)
+                    : "Build " + build.buildNum + " (failed)";
+
+            final List<JenkinsChange> changeSet = build.changes;
+
+            final List<String> changes;
+            if (changeSet.size() > 0)
+            {
+                changes = getChangelog(changeSet);
+            }
+            else
+            {
+                changes = Collections.singletonList("No git commits assigned");
+            }
+
+            changelogs.add(new Changelog(title, changes, build.getUrl()));
+        }
+        return changelogs;
+    }
+
+    private List<String> getChangelog(List<JenkinsChange> changeSet)
+    {
```

Comment:
could probably be made static

Reply:
Requires the instance variable githubUrl, so no static avail without extra param